File: stack/ts/stack.ts
class Stack<T> {
  private size: number;
  private stack: Array<T> = [];

  constructor(size: number) {
    this.size = size;
  }

  isEmpty(): boolean {
    return this.stack.length === 0;
  }

  isFull(): boolean {
    return this.stack.length === this.size;
  }

  length(): number {
    return this.stack.length;
  }

  push(item: T) {
    if (this.isFull()) {
      throw new Error("Stack is full.");
    }
  }

  pop(): T {
    if (this.isEmpty()) {
      throw new Error("Stack is empty.");
    }

    let poppedItem = this.stack.pop() as T;
    return poppedItem;
  }

  peek(): T {
    if (this.isEmpty()) {
      throw new Error("Stack is empty.");
    }

    let lastItemIndex: number = this.stack.length - 1;
    let item: T = this.stack[lastItemIndex];
    return item;
  }
}

const stack = new Stack<number>(5);
stack.push(6);
stack.push(9);
console.log(stack.pop());
console.log(stack.peek());
console.log(stack.length());

File: stack/swift/Stack.swift
import Foundation

class Stack<T> {
    private var size: Int
    private var stack: [T] = []
    
    init(size: Int) {
        self.size = size
    }
    
    enum StackError: Error {
        case overflow
        case underflow
    }
    
    var isEmpty: Bool {
        return stack.isEmpty
    }

    var isFull: Bool {
        return stack.count >= size
    }
    
    var length: Int {
        return stack.count
    }
    
    func push(item: T) throws {
        if isFull {
            throw StackError.overflow
        }
        stack.append(item)
    }
    
    func pop() throws -> T {
        if isEmpty {
            throw StackError.underflow
        }
        return stack.removeLast()
    }
    
    func peek() throws -> T {
        if isEmpty {
            throw StackError.underflow
        }
        return stack.last!
    }
}

let stack = Stack<Int>(size: 8)
try? stack.push(item: 4)
try? stack.push(item: 8)
try? stack.push(item: 16)
print(try! stack.peek())
print(try! stack.pop())
print(stack.length)
